src/TestData.ts

```diff
@@ -112,17 +113,36 @@ export class TestFile {
   }
 }
 
-function getFullPattern(start: TestDescribe | TestCase): string {
-  const parents: TestDescribe[] = []
-  let iter = start.parent
-  while (iter && iter instanceof TestDescribe) {
-    parents.push(iter)
-    iter = iter.parent
+class TaskName {
+  private readonly pattern: string
+  constructor(readonly start: TestCase | TestDescribe) {
+    this.pattern = transformTestPattern({ testName: start.name, isEach: start.isEach })
+  }
+
+  asVitestArgs(): string {
+    if (this.start instanceof TestCase)
+      return `^${this.join()}$`
+    else
+      return `^${this.join()}`
   }
 
-  parents.reverse()
-  if (parents.length)
-    return parents.reduce((a, b) => `${a + b.pattern} `, '') + start.pattern
-  else
-    return start.pattern
+  asFullMatchPattern(): string {
+    return `^${this.join()}$`
+  }
+
+  private join(): string {
+    const parents: (TestDescribe)[] = []
+    let iter: TestDescribe | TestFile | undefined = this.start.parent
+    while (iter && iter instanceof TestDescribe) {
+      parents.push(iter)
+      iter = iter.parent
+    }
+
+    parents.reverse()
+    // vitest's test task name starts with ' ' of root suite
+    if (parents.length)
+      return ` ${parents.reduce((a, b) => `${a + b.nameResolver.pattern} `, '')}${this.start.nameResolver.pattern}`
+    else
+      return ` ${this.start.nameResolver.pattern}`
```

Comment:
This mysterious leading space is considered a bug, but currently Vitest is keeping it to not break intellij:
- https://github.com/vitest-dev/vitest/issues/4045
- https://github.com/vitest-dev/vitest/issues/4103

To suppose both now and when it gets removed, maybe using `\s?` here would work?

Reply:
Good idea.